lib/cep/sources/correios.ex

```diff
@@ -11,14 +11,18 @@ defmodule Cep.Sources.Correios do
       {:ok, %HTTPoison.Response{status_code: 200, body: body}} ->
         {:ok, ResultFormatter.format(body)}
 
-      {:ok, %HTTPoison.Response{status_code: 500, headers: headers, body: body}} ->
+      {:ok, %HTTPoison.Response{status_code: 500, headers: _headers, body: body}} ->
         cond do
           Errors.cep_not_found?(body) -> cep_not_found_error()
           Errors.invalid_cep?(body) -> cep_not_found_error()
           true -> unknown_error(body)
         end
-      {:error, %HTTPoison.Error{reason: reason}} ->
-        {:error, reason}
+
+      {:ok, %HTTPoison.Response{status_code: code}} when code >= 400 and code <= 499 ->
+        cep_not_found_error()
```

Comment:
Why `cep_not_found_error()` for status codes that could be: forbidden, unauthorized, bad request and not found? I tested the Correios API some time ago and they don't return a 404 when the CEP is not found, for example. IMHO, these are all "unexpected"/unknown errors.

Reply:
You're right. I chose that response because it fit my domain, but thinking "opensourcely", it makes sense not to convert unknown things into cep not found. I'll change it.